src/shipment/controller.js

```diff
@@ -505,3 +510,323 @@ async function getPopulatedShipmentData(shipmentId = undefined) {
     return [e];
   }
 }
+
+async function getAsPdf(req, res) {
+  ExpressHandler(
+    async () => {
+      const { manifest, customer, dateCreated, shipmentId, deliveryMethod } = req.body;
+      const { title } = customer;
+
+      const orderNumbers = [];
+      for ( m of manifest ) {
+        if ( !orderNumbers.includes(m.orderNumber) ) orderNumbers.push(m.orderNumber);
+      }
+
+      const manifestBlocks = [];
+      const purchaseOrderNumbers = [];
+      const allShipmentsInfo = [];
+
+      for ( let idx = 0; idx < orderNumbers.length; idx ++) {
+        const orderNumber = orderNumbers[idx]
+        const promises = [];
+
+        promises.push( WorkOrder.findOne({ OrderNumber: orderNumber })
+          .lean()
+          .select('purchaseOrderNumber')
+          .exec()
+        );
+
+        promises.push(GetOrderFulfillmentInfo(orderNumber))
+
+        const [woDoc, [shippingError, shipmentInfo] ] = await Promise.all(promises);
+        const { purchaseOrderNumber } = woDoc;
+        purchaseOrderNumbers.push(purchaseOrderNumber);
+
+        if ( shippingError ) HTTPError('getting shipping contact information');
+        
+        allShipmentsInfo.push(shipmentInfo);
+
+        const _report = {};
+        const packingSlips = manifest.filter( x => x.orderNumber === orderNumber );
+        for ( const ps of packingSlips ) {
+          const { items } = ps;
+      
+          for (const i of items ) {
+            const { item, qty } = i;
+            const { _id } = item;
+            
+            if ( _id in _report === false ) {
+              _report[_id] = item;
+              _report[_id].qtyShipped = qty;
+            }
+            else {
+              _report[_id].qtyShipped += qty;
+            }
+          }
+        }
+
+        const items = [];
+        for ( const [ ,item] of Object.entries(_report) ) {
+          items.push(item);
+        }
+
+        const pageBreakAfter =  ( orderNumbers.length === 1 || idx === (orderNumbers.length - 1) ) ?
```

Comment:
You don't need a ternary operator here.
if you define a variable `const x = A || B` it's going to be assigned a boolean value.

If you negate your logic, you'll accomplish the same thing you're writing in an easier to ready manner

Reply:
This makes sense. Updated